interpreter/mesh/src/main/scala/io/buoyant/interpreter/mesh/Client.scala

```diff
@@ -105,6 +167,63 @@ object Client {
       case mesh.DelegateTreeRsp(Some(ptree)) => Some(fromBoundDelegateTree(ptree))
       case _ => None
     }
+
+    override def adminHandlers: Seq[Admin.Handler] = Seq(
+      Admin.Handler(
+        s"/interpreter_state/io.l5d.mesh${root.show}.json",
+        new ClientStateHandler
+      )
+    )
+
+    class ClientStateHandler extends Service[Request, Response] {
+
+      private[this] val mapper = Parser.jsonObjectMapper(Nil)
+
+      override def apply(request: Request): Future[Response] = {
+        val binds = bindCache
+        val resolves = resolveCache
+        val dtab = dtabCache
+
+        val bindState = binds.groupBy {
+          case ((dtab, path), bind) => path
+        }.map {
+          case (path, binds) =>
+            path.show -> binds.map {
+              case ((dtab, _), InstrumentedBind(act, stream)) =>
+                val snapshot = act.stateSnapshot().map { treeOpt =>
+                  treeOpt.map(_.show)
+                }
+                dtab.show -> Map(
+                  "state" -> snapshot,
+                  "watch" -> stream
+                )
+            }
+        }
+        val resolveState = resolves.map {
+          case (path, InstrumentedResolve(act, stream)) =>
+            path.show -> Map(
+              "state" -> act.stateSnapshot(),
+              "watch" -> stream
+            )
+        }
+        val dtabState = Option(dtab).map { dtab =>
```

Comment:
Is it possible for `dtab` to be `null`? if not, it would be clearer as `Some(...)`

Reply:
it is possible for dtab to be null.